proxy/proxy.go

```diff
@@ -134,12 +131,8 @@ func NewProxy(proxyHandler Handler, caKeyPath, caCertPath string, trustedCertifi
 	}
 	proxy.Verbose = log.GetLevel() == log.DebugLevel
 
-	// Handle HTTP requests with the specified handler.
-	p := proxy.OnRequest(goproxy.ReqHostMatches(regexp.MustCompile(httpRegexp)))
-	p.DoFunc(proxyHandler)
-
 	// Handle HTTPs requests with MITM and the specified handler.
-	p = proxy.OnRequest(goproxy.Not(goproxy.ReqHostMatches(regexp.MustCompile(httpRegexp))))
+	p := proxy.OnRequest()
 	if mitmHandler != nil {
 		p.HandleConnect(mitmHandler)
```

Comment:
Can we move the mitmhandler block above down to here, and merge it with this if statement.?